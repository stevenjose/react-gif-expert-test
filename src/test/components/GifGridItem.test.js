import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('Prueba en el componente GifGridItem', () => {

    const title = 'Jgla';
    const url = 'localhost/jgla.jpg'
    const wrapper = shallow( <GifGridItem title={title} url={url}/>);

    test('prueba en el componente <GifGridItem /> ', () => {

        expect( wrapper ).toMatchSnapshot();
    });

    test('debe contener title', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    });

    test('debe contener title', () => {
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
        
    });

    test('debe contener class', () => {
        const div = wrapper.find('div');
        expect( div.prop('className').includes('animate') ).toBe( true );
        
    });
    
});