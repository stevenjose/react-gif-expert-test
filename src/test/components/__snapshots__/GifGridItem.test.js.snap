// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el componente GifGridItem prueba en el componente <GifGridItem />  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Jgla"
    src="localhost/jgla.jpg"
  />
  <p>
     
    Jgla
     
  </p>
</div>
`;
